src/components/TodoList.js

```diff
@@ -0,0 +1,95 @@
+/* eslint-disable react/react-in-jsx-scope */
+import TodoNote from "./TodoNote";
+import "../index.css";
+import React, {useState, useEffect, useRef} from "react";
+import {nanoid} from "nanoid";
+import AddNote from "./AddNote";
+
+const TodoList = () => {
+  const getNotesFunction = () => {
+    const savedNotes = localStorage.getItem("react-notes-data");
+    return savedNotes ? JSON.parse(savedNotes) : [];
+  };
+
+  const [notes, setNotes] = useState(getNotesFunction);
+    const draggedOverNote = useRef();
+
+  const dragEnter = (noteToReplace) => {
+    draggedOverNote.note = noteToReplace;
+  };
+
+  const drop = (draggedNote) => {
+    let reorderedNotes = [...notes];
+    let draggedIndex = notes.indexOf(draggedNote);
+    let replacedIndex = notes.indexOf(draggedOverNote.note);
+    reorderedNotes[draggedIndex] = draggedOverNote.note;
+    reorderedNotes[replacedIndex] = draggedNote;
+    setNotes(reorderedNotes);
+  };
+
+  useEffect(() => {
+    window.addEventListener("beforeunload", handleBeforeUnload);
+    return () => {
+      window.removeEventListener("beforeunload", handleBeforeUnload);
+    };
+  }, []);
+
+  const handleBeforeUnload = (pageUnload) => {
+    pageUnload.preventDefault();
+    const message = "Are you sure you want to leave?";
+    pageUnload.returnValue = message;
+    return message;
+  };
+
+  const updateNotesFunction = () => {
+    localStorage.setItem("react-notes-data", JSON.stringify(notes));
```

Comment:
Why is this called update notes when it saves the notes in local storage?

Reply:
Done.